data-structures/1-5-MaximumInSlidingWindow.cpp

```diff
@@ -92,13 +92,11 @@ vector<uint> max_in_sliding_window(vector<uint> nums, uint size) {
 
   for (uint i = 0; i < nums.size(); ++i) {
     queue.enqueue(nums[i]);
-
-    if (i == size - 1) {
-      // at this moment we need to start to yield max
-      maxs.push_back(queue.max());
-    } else if (i >= size) {
-      queue.dequeue();
+    // condition can be replaced with queue.size() == size
```

Comment:
Don't understand this suggestion.

Reply:
Ahah, now I understand.

Reply:
In original code you have conditions like:
`
    if (i == size - 1) { ... } 
    else if (i >= size) { ...}
`

It is difficult to reason about this kind of conditions and easy to make off-by-one error

I suggest to void indexes comparison and use instead following code:

`
for (auto n : nums)
    q.push(n);
    if (q.size() == window_size ) { ...  }
`